core/tests/protractor_utils/ExplorationEditorSettingsTab.js

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Page object for the exploration editor's settings tab, for
+ * use in Protractor tests.
+ */
+
+var forms = require('./forms.js');
+var general = require('./general.js');
+
+var ExplorationEditorPage =
+  require('../protractor_utils/ExplorationEditorPage.js');
+
+var ExplorationEditorSettingsTab = function() {
+  /*
+   * Page objects
+   */
+  var explorationEditorPage = new ExplorationEditorPage.ExplorationEditorPage();
```

Comment:
This seems off to me -- the ExplorationEditorPage should contain the settings tab, not the other way around. Stuff like runFromSettingsTab() belongs in ExplorationEditorPage instead.

Reply:
main usage of `runFromSettingsTab()` is [within](https://github.com/oppia/oppia/blob/8573f34b51e7fb88045f0b3bc0dad88efedbc16a/core/tests/protractor_utils/ExplorationEditorSettingsTab.js#L66)  `ExplorationEditorSettingsTab.js`. 
If `runFromSettingsTab()` is put back to `ExplorationEditorPage` , I would still need to keep that line

Reply:
Addressed all comments with regards to SettingsTab PTAL!

Reply:
I don't think I understand the previous comment you had here. I was pointing out that runFromSettingsTab() should go in ExplorationEditorPage, but I'm not sure what you mean by "I would still need to keep that line".

I think though that maybe we should get rid of runFromSettingsTab() altogether and instead have the test do something like:

    explorationEditorPage.navigateToSettingsTab();
    settingsTab.doSomethingElse();

Reply:
I meant that I would still need access to `ExplorationEditorPage` inside `ExplorationEditorSettingsTab` due to `runFromSettingsTab` 
In order from SettingsTab to not requiring access to ExplorationEditorPage 's methods,  I just need to rewire it a little bit!

Reply:
Here is how one test suite from editorAndPlayer is using settings tab:
    
      creatorDashboardPage.clickCreateActivityButton();
      creatorDashboardPage.clickCreateExplorationButton();
      explorationEditorPage.exitTutorialIfNecessary();
      explorationEditorSettingsTab.setTitle(
        'Parent Exploration not in collection');
      explorationEditorSettingsTab.setCategory('Algebra');
      explorationEditorSettingsTab.setObjective('This is a parent exploration');

All function in explorationEditorSettings uses the callback mechanism provided by runFromSettingsTab: 
     
    this.runFromSettingsTab = function(callBackFunction) {
      this.navigateToSettingsTab();
      var result = callBackFunction();
      this.navigateToMainTab();
      return result;
    };
      
Are you saying we should get rid of `runFromSettingsTab` altogether and do something like:

     ...
     explorationEditorPage.navigateToSettingsTab();
     explorationEditorSettingsTab.setTitle(
        'Parent Exploration not in collection');
      explorationEditorSettingsTab.setCategory('Algebra');
      explorationEditorSettingsTab.setObjective('This is a parent exploration');
      explorationEditorPage.navigateToMainTab();

Reply:
Done!